### ISSUE TITLE
cpplint does not return anything.

### ISSUE THREAD

**Issue by yanghu** [scrooloose/syntastic#1488]
If i run `cpplint.py myfile.cpp` in cmd.exe window, it works normally.

But when I run syntastic check in VIM, it doesn't return any error.
I have cpplint.py's directory in my PATH.

According to [https://github.com/scrooloose/syntastic/issues/1031], I turned on syntastic debug, and here's the debug output.

```
syntastic: 43.274760: &shell = 'C:\Windows\system32\cmd.exe', &shellcmdflag = '/c', &shellpipe = '>%s 2>&1', &shellquote = '', &shellredir = '>%s 2>&1', &shellslash = 0, &shelltemp = 1, &s
hellxquote = '(', &shellxescape = '"&|<>()@^'
syntastic: 43.279051: UpdateErrors: cpplint
syntastic: 43.284008: CacheErrors: cpplint
syntastic: 43.289500: g:syntastic_aggregate_errors = 0
syntastic: 43.290880: getcwd() = 'D:\mysource'
syntastic: 43.292700: CacheErrors: Invoking checker: cpp/cpplint
syntastic: 43.295242: SyntasticMake: called with options: {'subtype': 'Style', 'errorformat': '%A%f:%l:  %m [%t],%-G%.%#', 'makeprg': 'cpplint.py --verbose=3 myfile.cpp', 'returns': [0,
1]}
syntastic: 43.710176: checker output: ['']
syntastic: 43.711761: raw loclist: []
syntastic: 43.712925: getLocList: checker cpp/cpplint returned 1
syntastic: 43.714048: cpp/cpplint raw: []
syntastic: 43.715203: quiet_messages filter: {}
syntastic: 43.716543: aggregated: {'_sorted': 0, '_name': '', '_owner': 3, '_columns': 1, '_rawLoclist': []}
```

Tried it for the second time after downloaded `cpplint.py` from google's site, got the same output like above, but without the last three lines. (the output ended at  `getLocList: checker cpp/cpplint returned 1`.

**Comment by lcd047** [comment]
Please upgrade syntastic to the current master HEAD, then open your test file, set `g:syntastic_debug` to 3, run the checker, run `:mes`, and post the output.

**Comment by yanghu** [comment]
thanks for the reply. 
that's what I did. my current HEAD is at 74586ff 

the text I pasted is the output. 

My cpplint.py is obtained from 
https://google-styleguide.googlecode.com/svn/trunk/cpplint/cpplint.py

**Comment by lcd047** [comment]
Ok.  Please open your test file, run `:SyntasticInfo`, and post the output.  Then run this command and post the output:
```vim
:echo system('cpplint.py --verbose=3 myfile.cpp')
```

**Comment by yanghu** [comment]
The environment is on my work computer. I'll try that next week and post the results here. thanks~

**Comment by yanghu** [comment]
`:SyntasticInfo` output is:

```
Syntastic version: 3.6.0-137 (Vim 704, Windows)
Info for filetype: cpp
Global mode: active
Filetype cpp is active
The current file will be checked automatically
Available checkers: clang_check clang_tidy cpplint gcc
Currently enabled checker: cpplint
```

run `:echo system('cpplint.py --verbose=3 myfile.cpp')` in VIM has no output. I changed pwd to the file's directory before running this command.

run `cpplint.py --verbose=3 myfile.cpp` in cmd.exe has following output

```
myfile.cpp:0:  No copyright message found.  You should have a line: "Copyright [year] <Copyright Owner>"  [legal/copyright] [5]
myfile.cpp:4:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
Done processing myfile.cpp
Total errors found: 2
```

**Comment by lcd047** [comment]
> run `:echo system('cpplint.py --verbose=3 myfile.cpp')` in VIM has no output. I changed pwd to the file's directory before running this command.

Well, this is what you need to fix then, as this is what syntastic uses under the hood.  But I'm afraid I have no idea how you could do that, sorry.  The problem is between Vim, Python, and Windows.  It has nothing to do with syntastic.  Solve it, and the syntastic checker should also start working.

**Comment by yanghu** [comment]
Thanks. I'll check it out. 
I tried  something like `echo system('echo %username%')` and seems both `echo` and `system` are working. 

what about these lines? are they just a consequence of the empty echo output? 
```
syntastic: 43.710176: checker output: ['']
syntastic: 43.711761: raw loclist: []
syntastic: 43.712925: getLocList: checker cpp/cpplint returned 1
syntastic: 43.714048: cpp/cpplint raw: []
syntastic: 43.715203: quiet_messages filter: {}
```

**Comment by yanghu** [comment]
If I use `system('python cpplint.py myfile.cpp')` instead of `system('cpplint.py myfile.cpp')`, I can get the output. 
Is there any where I can modify syntastic and tell it to invoke `python cpplint.py` instead of `cpplint.py`? 

maybe it's beacuse windows command line is not recognizing the hasbang line in cpplint.py?

**Comment by lcd047** [comment]
> Is there any where I can modify syntastic and tell it to invoke `python cpplint.py` instead of `cpplint.py`?

Yes, you can set:
```vim
let g:syntastic_cpp_cpplint_exe = 'python cpplint.py'
```
(please note the `exe`, not `exec`).

**Comment by yanghu** [comment]
thanks! Finally I got it working. 

1. directly change `cpplint.vim` file does not help, after I changed it to `python cpplint.py` it cannot find my cpplint checker. 

2. I set `let g:syntastic_cpp_cpplint_exe = 'python cpplint.py'`, but it didn't help. the error message is my python cannot find `cpplint.py`. I added its path to my PYTHONPATH, still won't work. 

3. Then I just hard coded my cppilnt.py's absolute path into the command, something like 

```
let g:syntastic_cpp_cpplint_exe = 'python D:\checkers\cpplint.py'
```

And it works.
